jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -37,13 +38,18 @@ public static void premain(String agentArgument, Instrumentation instrumentation
      new JmxCollector(new File(file)).register();
      DefaultExports.initialize();
 
-     server = new Server(socket);
+     server = new Server();
      QueuedThreadPool pool = new QueuedThreadPool();
      pool.setDaemon(true);
      pool.setMaxThreads(10);
      pool.setMaxQueued(10);
      pool.setName("jmx_exporter");
      server.setThreadPool(pool);
+     SelectChannelConnector connector = new SelectChannelConnector();
+     connector.setHost(socket.getHostName());
```

Comment:
Hmm, will this bind to only localhost if no host is provided as an arg?
This may also do a reverse DNS lookup, which doesn't seem right.

I think the arg parsing logic needs to be redone to remove the socket, and only call this method if needed.